### PR TITLE
Fix 3216: queue the VoiceInstruction so it just play one at a time.

### LINKED ISSUE #3216: Voice instructions is overlapping
When running the `InstructionViewActivity` in the `example` app, noticed the same voice instructions were played multiple times.

Update: The issue was not about duplicate, but actually it's overlapping. The second voice instruction will be played at the same time while the first voice instruction is still playing. It will be fixed by #3297

### PR DESCRIPTION
**Pull request by JunDai**
## Description

Fix #3216: queue the VoiceInstruction so it just play one at a time.

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [ ] I have added the appropriate milestone and project boards

### Goal

To improve the voice instruction quality.

### Implementation

Queue the voice instructions, and only play it when the speechPlayer is ready to play.

## Testing

Please describe the manual tests that you ran to verify your changes

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [ ] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR
- [ ] We might need to update / push `api/current.txt` files after running `$> make 1.0-core-update-api` (Core) / `$> make 1.0-ui-update-api` (UI) if there are changes / errors we're 🆗 with (e.g. `AddedMethod` changes are marked as errors but don't break SemVer) 🚀 If there are SemVer breaking changes add the `SEMVER` label. See [Metalava](https://github.com/mapbox/mapbox-navigation-android/blob/master/docs/metalava.md) docs

### CODE REVIEW

**Comment by RingerJK** [comment]
@JunDai looks nice! Could you add `@Null`/`@Nonull` annotations to java code and add tests?

**Comment by JunDai** [comment]
> @JunDai looks nice! Could you add `@Null`/`@Nonull` annotations to java code and add tests?

thanks! Please check again.

**Comment by JunDai** [comment]
@Guardiola31337 - Can you also please verify and confirm if this PR still looks good and fixes the issue?

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/voice/SpeechPlayerProvider.java [hunk 1]
```diff
@@ -31,6 +31,8 @@
   private List<SpeechPlayer> speechPlayers = new ArrayList<>(2);
   private VoiceInstructionLoader voiceInstructionLoader;
   private ConnectivityStatusProvider connectivityStatus;
+  private SpeechPlayerState speechPlayerState = SpeechPlayerState.IDLE;
+  private SpeechPlayerStateChangeObserver observer = null;
```
Comment (RingerJK) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/voice/SpeechPlayerProvider.java:35]:
```suggestion
  @Nullable
  private SpeechPlayerStateChangeObserver observer = null;
```

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/voice/SpeechPlayerProvider.java [hunk 2]
```diff
@@ -31,6 +31,8 @@
   private List<SpeechPlayer> speechPlayers = new ArrayList<>(2);
   private VoiceInstructionLoader voiceInstructionLoader;
   private ConnectivityStatusProvider connectivityStatus;
+  private SpeechPlayerState speechPlayerState = SpeechPlayerState.IDLE;
```
Comment (RingerJK) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/voice/SpeechPlayerProvider.java:34]:
```suggestion
  @Nonnull
  private SpeechPlayerState speechPlayerState = SpeechPlayerState.IDLE;
```

**Pull request merged**